src/greatwall/resources/knowledge/fractal.py

```diff
@@ -33,11 +33,30 @@ def __init__(
         self._image_pixels: Optional[np.array] = None
 
     def get_valid_parameter_from_value(self, value: Union[bytes, bytearray]):
-        parameter = "2."
-        for byte_value in list(value)[:4]:
-            parameter += str(byte_value)
-
-        return float(parameter)
+        # NOTE: We inverting the order of digits by operation [::-1] on string,
+        # to minimize Benford's law bias.
+        real_p = (
+            "2."
+            + str(
+                int.from_bytes(
+                    bytes(list(value)[0 : int(len(value) / 2)])
+                    + "real_p".encode(encoding="utf-8"),
```

Comment:
I think this should use some divide-and-conquer. Namely: to have the hashing of state appended with a flag like 'real_p' or 'imaginary_p' be done in a function to avoid "code relation". By code relation, I mean have two distinct pieces of code bearing an analogy or even equality which overburdens maintenance and readability.